### ISSUE TITLE
CI: K8sServicesTest Checks service across nodes with L7 policy Tests NodePort with L7 Policy: Exitcode: 42

### ISSUE THREAD

**Issue by pchaigno** [cilium/cilium#13011]
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/3458/testReport/Suite-k8s-1/13/K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy/
[612b41a5_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip](https://github.com/cilium/cilium/files/5144042/612b41a5_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip)

Happened in backport PR #12990 which doesn't appear to have any related changes. I couldn't find any other occurrence in CI dashboard that wasn't due to a previous test failing (`BeforeAll`).

### Stacktrace
```
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-K8s/1.13-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:514
Request from k8s1 to service tftp://192.168.36.11:31018/hello failed
Expected command: kubectl exec -n kube-system log-gatherer-rxfzh -- /bin/bash -c 'fails=""; id=$RANDOM; for i in $(seq 1 10); do if curl --path-as-is -s -D /dev/stderr --fail --connect-timeout 5 --max-time 20 tftp://192.168.36.11:31018/hello -H "User-Agent: cilium-test-$id/$i"; then echo "Test round $id/$i exit code: $?"; else fails=$fails:$id/$i=$?; fi; done; if [ -n "$fails" ]; then echo "failed: $fails"; fi; cnt="${fails//[^:]}"; if [ ${#cnt} -gt 0 ]; then exit 42; fi' 
To succeed, but it failed:
Exitcode: 42 
Err: Process exited with status 42
Stdout:
 	 
	 Hostname: testds-nw5sj
	 
	 Request Information:
	 	client_address=10.0.1.168
	 	client_port=41842
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 11784/1 exit code: 0
	 
	 Hostname: testds-nw5sj
	 
	 Request Information:
	 	client_address=10.0.1.168
	 	client_port=42269
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 11784/2 exit code: 0
	 
	 Hostname: testds-n2h9z
	 
	 Request Information:
	 	client_address=10.0.2.15
	 	client_port=43873
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 11784/3 exit code: 0
	 
	 Hostname: testds-n2h9z
	 
	 Request Information:
	 	client_address=10.0.2.15
	 	client_port=43052
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 11784/4 exit code: 0
	 
	 Hostname: testds-nw5sj
	 
	 Request Information:
	 	client_address=10.0.1.168
	 	client_port=40734
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 11784/5 exit code: 0
	 
	 Hostname: testds-n2h9z
	 
	 Request Information:
	 	client_address=10.0.2.15
	 	client_port=37170
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 11784/6 exit code: 0
	 
	 Hostname: testds-nw5sj
	 
	 Request Information:
	 	client_address=10.0.1.168
	 	client_port=60738
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 11784/8 exit code: 0
	 
	 Hostname: testds-nw5sj
	 
	 Request Information:
	 	client_address=10.0.1.168
	 	client_port=51208
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 11784/9 exit code: 0
	 
	 Hostname: testds-nw5sj
	 
	 Request Information:
	 	client_address=10.0.1.168
	 	client_port=45831
	 	real path=/hello
	 	request_scheme=tftp
	 
	 Test round 11784/10 exit code: 0
	 failed: :11784/7=71
	 
Stderr:
 	 command terminated with exit code 42
	 

/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-K8s/1.13-gopath/src/github.com/cilium/cilium/test/k8sT/Services.go:744
```
### Standard Output
```
Number of "context deadline exceeded" in logs: 0
Number of "level=error" in logs: 0
Number of "level=warning" in logs: 0
Number of "Cilium API handler panicked" in logs: 0
Number of "Goroutine took lock for more than" in logs: 0
No errors/warnings found in logs
Cilium pods: [cilium-99l2z cilium-bkc69]
Netpols loaded: 
CiliumNetworkPolicies loaded: default::l7-policy-demo 
Endpoint Policy Enforcement:
Pod   Ingress   Egress
Cilium agent 'cilium-99l2z': Status: Ok  Health: Ok Nodes "" ContinerRuntime:  Kubernetes: Ok KVstore: Ok Controllers: Total 48 Failed 0
Cilium agent 'cilium-bkc69': Status: Ok  Health: Ok Nodes "" ContinerRuntime:  Kubernetes: Ok KVstore: Ok Controllers: Total 28 Failed 0
```

**Comment by tklauser** [comment]
Hit on #14718 https://jenkins.cilium.io/job/Cilium-PR-K8s-1.20-kernel-4.9/496/

**Comment by gandro** [comment]
Hit on #14739: https://jenkins.cilium.io/job/Cilium-PR-K8s-1.20-kernel-4.9/539/testReport/junit/Suite-k8s-1/20/K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy/

**Comment by ungureanuvladvictor** [comment]
Hit this on https://jenkins.cilium.io/job/Cilium-PR-K8s-1.21-kernel-4.9/946/testReport/junit/Suite-k8s-1/21/K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy/. 
[b6df4677_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip](https://github.com/cilium/cilium/files/6751619/b6df4677_K8sServicesTest_Checks_service_across_nodes_with_L7_policy_Tests_NodePort_with_L7_Policy.zip)

**Comment by brb** [comment]
It should be fixed by #12511.
